compiler/rustc_lint/src/types.rs

```diff
@@ -136,6 +136,13 @@ fn lint_overflowing_range_endpoint<'tcx>(
     expr: &'tcx hir::Expr<'tcx>,
     ty: &str,
 ) -> bool {
+    let (expr, cast_ty) = if let Node::Expr(par_expr) = cx.tcx.hir().get(cx.tcx.hir().parent_id(expr.hir_id))
```

Comment:
Can you add a comment here describing what this if does? Something like
```rust
// Look past casts to support cases like `0..256 as u8`
```

Reply:
Of course!